tests/examples.rs

```diff
@@ -8,7 +8,7 @@ fn eval_file(file: &str) -> Result<RichTerm, Error> {
     let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
     path.push(format!("examples/{}", file));
     let mut p = Program::new_from_file(path).expect("could not load file as a program");
-    p.eval()
+    p.eval_full()
```

Comment:
`eval_full()` does more than we want, as it will also substitute all the variables inside the resulting term. It's maybe not immensely important for now given the size and number of examples, but it's probably better to save potential future compute power. What we want is basically to do everything that's done in `eval_full` but without the last step. It may make sense to factorize one part of `eval_full` as `eval_deep`, if that avoids duplicating code.

Reply:
Done